vm/option.go

```diff
@@ -13,20 +16,38 @@ import (
 
 type RegisterFunc func(*VM)
 
-type OptionFunc func(*VM, []byte) error
+type OptionFunc func(vm *VM, configBytes []byte) error
+
+type OptionConfigFunc[T any] func(vm *VM, config T) error
 
 type Option struct {
 	Namespace  string
 	OptionFunc OptionFunc
 }
 
-func NewOption(namespace string, optionFunc OptionFunc) Option {
+func NewOptionWithBytes(namespace string, optionFunc OptionFunc) Option {
```

Comment:
Looking to see if we can remove this so we can just have a single `NewOption` function signature out of this package since I find this naming confusing - ref earlier comment.